### PR TITLE
Convert more tests

### PR DESCRIPTION
**Pull request by richsalz**
ct_test,evp_extra_test,wpackettest,packettest
Add strncmp TEST wrappers
And make some style/consistency fixes to ct_test

### CODE REVIEW

**Comment by dot-asm** [comment]
Red cross from travis is relevant.

**Comment by pauli-oracle** [comment]
The error message prints from `test_strn_eq` and `test_strn_ne` will fail if the strings aren't zero terminated.  The `print_string_maybe_null` are the problem.

File: test/testutil.h [hunk 1]
```diff
@@ -292,6 +300,7 @@ void test_info_c90(const char *desc, ...) PRINTF_FORMAT(1, 2);
 # define TEST_ptr_null(a)     test_ptr_null(__FILE__, __LINE__, #a, a)
 
 # define TEST_str_eq(a, b)    test_str_eq(__FILE__, __LINE__, #a, #b, a, b)
+# define TEST_strn_eq(a, b, n) test_strn_eq(__FILE__, __LINE__, #a, #b, a, b, n)
 # define TEST_str_ne(a, b)    test_str_ne(__FILE__, __LINE__, #a, #b, a, b)
```
Comment (levitte) [inline test/testutil.h]: You should add a macro `TEST_strn_ne` as well.

File: test/packettest.c [hunk 2]
```diff
@@ -8,328 +8,311 @@
...
+            || !TEST_char_eq(bytes[1], 4)
+            || !TEST_char_eq(bytes[2], 6)
+            || !TEST_char_eq(bytes[3], 8)
+            || !TEST_size_t_eq(PACKET_remaining(&pkt), BUF_LEN - 4)
+            || !TEST_true(PACKET_forward(&pkt, BUF_LEN - 8))
+            || !TEST_true(PACKET_copy_bytes(&pkt, bytes, 4))
+            || !TEST_char_eq(bytes[0], 0xf8)
+            || !TEST_char_eq(bytes[1], 0xfa)
+            || !TEST_char_eq(bytes[2], 0xfc)
+            || !TEST_char_eq(bytes[3], 0xfe)
```
Comment (levitte) [inline test/packettest.c]: In the four cases above, use `TEST_uchar_eq` instead.

File: test/testutil.h (hunk 1 above)
Comment (richsalz) [inline test/testutil.h]: Added ```TEST_strn_ne```

File: test/packettest.c (hunk 2 above)
Comment (richsalz) [inline test/packettest.c]: fixed.

**Comment by richsalz** [comment]
Added precision args to the strn printf's.
Additional commit pushed.

**Comment by levitte** [comment]
Travis errors are relevant

**Comment by richsalz** [comment]
I think it's a gcc bug, but just pushed a workaround for it.

**Comment by richsalz** [comment]
adcd8e3 in master, thanks!

**Pull request closed without merging**